Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -4362,6 +4362,29 @@ void MediaPlayerPrivateGStreamer::checkPlayingConsistency()
     }
 }
 
+static void applyAudioSinkDevice(GstElement* audioSinkBin, const String& deviceId)
+{
+    for (auto* element : GstIteratorAdaptor<GstElement>(GUniquePtr<GstIterator>(gst_bin_iterate_sinks(GST_BIN_CAST(audioSinkBin))))) {
+        // pulsesink and alsasink have a "device" property, whilst pipewiresink has "target-object"
+        if (GST_IS_AUDIO_BASE_SINK(element) && gstObjectHasProperty(element, "device"))
```

Comment:
This is a bit fragile, it might make sense to check the element factory name instead. But it's ok, we use this `gstObjectHasProperty()` approach elsewhere, it can be improved as a follow-up.

Reply:
Indeed I tried to do what I could see in other parts of the code, but indeed the check was looking a bit too lax and that's why I added the `GST_IS_*BASE_SINK()` condition as well.

You have a good point, though, so I have filed https://bugs.webkit.org/show_bug.cgi?id=287017 to improve the checks later on.